tests/unit/source_code/python/test_python_ast.py

```diff
@@ -352,3 +407,86 @@ def test_renumber_fails() -> None:
 
 def test_const_is_not_from_module() -> None:
     assert not Tree(Const("xyz")).is_from_module("spark")
+
+
+class DummyPythonLinter(PythonLinter):
+    """Dummy python linter yielding dummy advices for testing purpose."""
+
+    def lint_tree(self, tree: Tree) -> Iterable[Advice]:
+        yield Advice("dummy", "dummy advice", 0, 0, 0, 0)
+        yield Advice("dummy", "dummy advice", 1, 1, 1, 1)
+
+
+def test_dummy_python_linter_lint_lints_tree() -> None:
+    linter = DummyPythonLinter()
+    advices = list(linter.lint("print(1)"))
+    assert advices == [Advice("dummy", "dummy advice", 0, 0, 0, 0), Advice("dummy", "dummy advice", 1, 1, 1, 1)]
+
+
+def test_dummy_python_linter_lint_yields_failure_due_to_parse_error() -> None:
+    linter = DummyPythonLinter()
+    advices = list(linter.lint("print(1"))  # Closing parenthesis is missing on purpose
+    assert advices == [
+        Failure(
+            code="python-parse-error",
+            message="Failed to parse code due to invalid syntax: print(1",
+            start_line=0,
+            start_col=5,
+            end_line=0,
+            end_col=1,
+        )
+    ]
+
+
+def test_python_sequential_linter_lint_lints_tree() -> None:
+    linter = PythonSequentialLinter([DummyPythonLinter()], [], [])
+    advices = list(linter.lint("print(1)"))
+    assert advices == [Advice("dummy", "dummy advice", 0, 0, 0, 0), Advice("dummy", "dummy advice", 1, 1, 1, 1)]
+
+
+class NodeGlobalsLinter(PythonLinter):
+    """Get the node globals"""
+
+    def lint_tree(self, tree: Tree) -> Iterable[Advice]:
+        globs = ""
+        if isinstance(tree.node, Module):
+            globs = ",".join(tree.node.globals.keys())
+        yield Advice("globals", globs, 0, 0, 0, 0)
+
+
+def test_python_sequential_linter_lint_has_no_globals() -> None:
+    linter = PythonSequentialLinter([NodeGlobalsLinter()], [], [])
+    advices = list(linter.lint("print(1)"))
+    assert advices == [Advice("globals", "", 0, 0, 0, 0)]
+
+
+def test_python_sequential_linter_lint_has_one_global() -> None:
+    linter = PythonSequentialLinter([NodeGlobalsLinter()], [], [])
+    advices = list(linter.lint("a = 1"))
+    assert advices == [Advice("globals", "a", 0, 0, 0, 0)]
+
+
+def test_python_sequential_linter_lint_has_two_globals() -> None:
+    linter = PythonSequentialLinter([NodeGlobalsLinter()], [], [])
+    advices = list(linter.lint("a = 1;b = 2"))
+    assert advices == [Advice("globals", "a,b", 0, 0, 0, 0)]
+
+
+def test_python_sequential_linter_lint_is_stateless() -> None:
+    """Globals from previous lint calls should not be part of later calls"""
+    linter = PythonSequentialLinter([NodeGlobalsLinter()], [], [])
+    list(linter.lint("a = 1"))
+    advices = list(linter.lint("b = 2"))
+    assert advices == [Advice("globals", "b", 0, 0, 0, 0)]
+
+
+def test_python_sequential_linter_extend_globals() -> None:
```

Comment:
I ended my day with this unit test, I am not sure if this is the intention of the `append_globals` on the `PythonSequentialLinter` but it looks like they do not end up at the `Tree`. Note: it might also be that I am looking at the wrong node and that they end up in the tree after all.